### PR TITLE
Add inner execution queue to TestRun (fixes #567)

### LINKED ISSUE #567: Implement action error recovery
``` js
try {
    await t.click('#error');
} catch(e) {
}

await t.click('#btn');
```

### PR DESCRIPTION
**Pull request by inikulin**
Thanks to @AlexanderMoskovkin for assistance.

\cc @helen-dikareva @AndreyBelym

### CODE REVIEW

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

**Comment by AlexanderMoskovkin** [comment]
The `Should raise an error on an attempt to execute a new command while the previous command execution is not yet finished` server test is failed. It's obsolete since you've implemented a command queue in the `test-run`

**Comment by inikulin** [comment]
@AlexanderMoskovkin Yes, but take a look at first functional tests fail - same happened on my machine - legacy API `waitFor` test hangs up (looks like in FF or IE), because chrome-only tests afterwards works just fine.

**Comment by AlexanderMoskovkin** [comment]
It shouldn't be an effect of your changes, because you haven't modified legacy API related code, but sure, we should take an attention on this

**Comment by AlexanderMoskovkin** [comment]
the test hangs in FF, we will investigate this (#568) @georgiy-abbasov please take a look at this when you have time

**Comment by AlexanderMoskovkin** [comment]
just a mobile client test failed, I've restarted it

**Comment by inikulin** [comment]
FPR

File: src/test-run/index.js [hunk 1]
```diff
@@ -40,11 +40,12 @@ export default class TestRun extends Session {
...
-        this.pendingRequest            = null;
-        this.pendingJsError            = null;
-        this.currentCommandCallsite    = null;
-        this.pendingWindowManipulation = null;
+        this.driverTaskQueue          = [];
+        this.browserManipulationQueue = [];
+        this.testDoneCommandQueued    = false;
+
+        this.pendingRequest           = null;
+        this.pendingPageError         = null;
```
Comment (inikulin) [inline src/test-run/index.js]: https://github.com/inikulin/testcafe/blob/fb8f2e00475adb5d878d045d7e04177ff0aec0d0/src/test-run/index.js#L174 - nope

Comment (helen-dikareva) [inline src/test-run/index.js]: oh, I see
lgtm

**Comment by AndreyBelym** [comment]
lgtm

**Pull request merged**